torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -644,6 +644,12 @@ bool ProcessGroupNCCL::WorkNCCL::wait(std::chrono::milliseconds timeout) {
       static_cast<int>(devices_.size())); // worldSize
   synchronizeInternal(timeout);
   // Always return true, because abort API is not implemented.
+  if (debug_level() == DebugLevel::Detail) {
+    auto numel = getTensorsNumel(*outputs_);
+    auto hashValue = hashTensors(*outputs_);
+    PRINT_COLLECTIVE_HASH_SIGNATURE(
+        "launching", rank_, opTypeToString(opType_), numel, hashValue);
```

Comment:
and this one should be "after" or "output" -- becuase as @fegin pointed out, `hashTensors()` will make a cuda dtoh sync and produce the real value, so it is not how i originally understood.